news/11342.feature.rst

```diff
@@ -0,0 +1,2 @@
+Add a version check to the ``__pip-runner__.py`` script, to make error handling
+more robust for the ``--python`` option.
```

Comment:
Perhaps we should have a single news entry for the `--python` option, assuming this goes into the same release as the option.

Reply:
I'm fine with that, in which case we can simply omit the news item from this PR.

Reply:
I'd omit this news entry, yea.

Reply:
Done.